src/Sunset/EventListener/AddSunsetHeaderListener.php

```diff
@@ -0,0 +1,45 @@
+<?php
+
+/*
+ * This file is part of the API Platform project.
+ *
+ * (c) Kévin Dunglas <dunglas@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace ApiPlatform\Core\Sunset\EventListener;
+
+use ApiPlatform\Core\Metadata\Resource\Factory\ResourceMetadataFactoryInterface;
+use ApiPlatform\Core\Util\RequestAttributesExtractor;
+use Symfony\Component\HttpKernel\Event\FilterResponseEvent;
+
+class AddSunsetHeaderListener
+{
+    private $resourceMetadataFactory;
+
+    public function __construct(ResourceMetadataFactoryInterface $resourceMetadataFactory)
+    {
+        $this->resourceMetadataFactory = $resourceMetadataFactory;
+    }
+
+    public function onKernelResponse(FilterResponseEvent $event)
+    {
+        $request = $event->getRequest();
+        $attributes = RequestAttributesExtractor::extractAttributes($request);
+
+        if (!$attributes) {
+            return;
+        }
+
+        $response = $event->getResponse();
+        $metadata = $this->resourceMetadataFactory->create($attributes['resource_class']);
+
+        if (null !== $metadata->getAttribute('sunset')) {
```

Comment:
You should use the following:

```php
if ($sunset = $metadata->getOperationAttribute($attributes, 'sunset', null, true)) {
  $response->headers->set('Sunset', (new \DateTimeImmutable($sunset)->format(\DateTime::RFC1123));
}
```

It will allow to define sunset headers only for some operations, or to define different dates for different operations.

Also, the trick with `DateTime` allows the user to set the date in any format (even computations like "2019/01/01 + 1 month" will work), but to always output the valid format.

Reply:
Wow. That is actually very smart! Thanks!

Reply:
Done